test/test_navierstokes.py

```diff
@@ -0,0 +1,375 @@
+            f"rhoe_rhs = {rhoe_rhs}\n"
+            f"rhov_rhs = {rhov_rhs}\n"
+            f"rhoy_rhs = {rhoy_rhs}\n"
+        )
+
+        assert actx.to_numpy(discr.norm(rho_resid, np.inf)) < tolerance
+        assert actx.to_numpy(discr.norm(rhoe_resid, np.inf)) < tolerance
+        for i in range(dim):
+            assert actx.to_numpy(discr.norm(mom_resid[i], np.inf)) < tolerance
+        for i in range(nspecies):
+            assert actx.to_numpy(discr.norm(rhoy_resid[i], np.inf)) < tolerance
+
+        err_max = actx.to_numpy(discr.norm(rho_resid, np.inf))
+        eoc_rec0.add_data_point(1.0 / nel_1d, err_max)
+
+        # set a non-zero, but uniform velocity component
+        for i in range(len(mom_input)):
+            mom_input[i] = discr.zeros(actx) + (-1.0) ** i
+
+        cv = make_conserved(
+            dim, mass=mass_input, energy=energy_input, momentum=mom_input,
+            species_mass=species_mass_input)
+
+        state = make_fluid_state(gas_model=gas_model, cv=cv)
+        boundaries = {BTAG_ALL: DummyBoundary()}
+        ns_rhs = ns_operator(discr, gas_model=gas_model, boundaries=boundaries,
+                             state=state, time=0.0)
+
+        rhs_resid = ns_rhs - expected_rhs
+
+        rho_resid = rhs_resid.mass
+        rhoe_resid = rhs_resid.energy
+        mom_resid = rhs_resid.momentum
+        rhoy_resid = rhs_resid.species_mass
+
+        assert actx.to_numpy(discr.norm(rho_resid, np.inf)) < tolerance
+        assert actx.to_numpy(discr.norm(rhoe_resid, np.inf)) < tolerance
+
+        for i in range(dim):
+            assert actx.to_numpy(discr.norm(mom_resid[i], np.inf)) < tolerance
+        for i in range(nspecies):
+            assert actx.to_numpy(discr.norm(rhoy_resid[i], np.inf)) < tolerance
+
+        err_max = actx.to_numpy(discr.norm(rho_resid, np.inf))
+        eoc_rec1.add_data_point(1.0 / nel_1d, err_max)
+
+    logger.info(
+        f"V == 0 Errors:\n{eoc_rec0}"
+        f"V != 0 Errors:\n{eoc_rec1}"
+    )
+
+    assert (
+        eoc_rec0.order_estimate() >= order - 0.5
+        or eoc_rec0.max_error() < 1e-9
+    )
+    assert (
+        eoc_rec1.order_estimate() >= order - 0.5
+        or eoc_rec1.max_error() < 1e-9
+    )
+
+
+# Box grid generator widget lifted from @majosm and slightly bent
+def _get_box_mesh(dim, a, b, n, t=None):
+    dim_names = ["x", "y", "z"]
+    bttf = {}
+    for i in range(dim):
+        bttf["-"+str(i+1)] = ["-"+dim_names[i]]
+        bttf["+"+str(i+1)] = ["+"+dim_names[i]]
+    from meshmode.mesh.generation import generate_regular_rect_mesh as gen
+    return gen(a=a, b=b, n=n, boundary_tag_to_face=bttf, mesh_type=t)
+
+
+@pytest.mark.parametrize("order", [2, 3])
+def test_poiseuille_rhs(actx_factory, order):
+    """Test the Navier-Stokes operator using a Poiseuille state.
+
+    This state should yield rhs = 0 to FP.  The test is performed for 1, 2,
+    and 3 dimensions, with orders 1, 2, and 3, with and without passive species.
+    """
```

Comment:
Some of the details/analysis from https://notes.tiker.net/IQb-yrlFT82tKscEFkhrxA should probably make it in here.